search-provider/eks-discovery-feed-provider.c

```diff
@@ -181,19 +183,80 @@ underscorify (const gchar *string)
 }
 
 static void
+add_key_value_pair_to_variant (GVariantBuilder *builder,
+                               const char      *key,
+                               const char      *value)
+{
+  g_variant_builder_open (builder, G_VARIANT_TYPE ("{ss}"));
+  g_variant_builder_add (builder, "s", key);
+  g_variant_builder_add (builder, "s", value);
+  g_variant_builder_close (builder);
+}
+
+static void
 add_key_value_pair_from_model_to_variant (EkncContentObjectModel *model,
                                           GVariantBuilder        *builder,
                                           const char             *key)
 {
-  g_variant_builder_open (builder, G_VARIANT_TYPE ("{ss}"));
   g_autofree gchar *value = NULL;
   g_autofree gchar *underscore_key = underscorify (key);
   g_object_get (model, key, &value, NULL);
-  g_variant_builder_add (builder, "s", underscore_key);
-  g_variant_builder_add (builder, "s", value);
-  g_variant_builder_close (builder);
+  add_key_value_pair_to_variant (builder, underscore_key, value);
+}
+
+static GPtrArray *
+variant_array_string_to_ptr_array (GVariant *variant,
+                                   unsigned int reserved_size)
+{
+  GVariantIter iter;
+  gchar *value;
+
+  GPtrArray *array = g_ptr_array_new_full (reserved_size, g_free);
+
+  g_variant_iter_init (&iter, variant);
+  while (g_variant_iter_next (&iter, "s", &value))
+    {
+      g_ptr_array_add (array, value);
+      /* No need to free value here, it is transferred to
+       * array */
+    }
+
+  return array;
 }
 
+static GPtrArray *
+variant_array_variant_to_ptr_array (GVariant *variant,
+                                    unsigned int reserved_size)
+{
+  GVariantIter iter;
+  GVariant *value;
+
+  GPtrArray *array = g_ptr_array_new_full (reserved_size, g_free);
+
+  g_variant_iter_init (&iter, variant);
+  while (g_variant_iter_loop (&iter, "v", &value))
+    {
+      g_ptr_array_add (array, g_strdup(g_variant_get_string(value, NULL)));
```

Comment:
You could use `g_variant_dup_string()`, but as I note below, I think this function should go away.